fallback_builtins.h

```diff
@@ -9,6 +9,30 @@
 #  include "arch/x86/x86.h"
 #endif
 
+#ifdef __AVX2__
+/* Unfortunately GCC _and_ clang didn't support these things until version
+ * 10 and 12, respectively */
+#include <immintrin.h>
+
+#if (defined(__GNUC__) && __GNUC__ < 10) || (defined(__clang__) && __clang__ < 12)
```

Comment:
clang also defines `__GNUC__`, so the first test would always be true for clang, as it defines `__GNUC__` as `4`

Reply:
Ugh, of course they do :-/.  Alright, I'll flip those around.